src/interface/src/app/services/plan.service.ts

```diff
@@ -71,12 +92,21 @@ export class PlanService {
     this.planState$.next(updatedState);
   }
 
-  /** Temporary stub for the api. */
-  private createPlanMockApi(plan: BasePlan): Plan {
-    this.tempPlanId = ++this.tempPlanId;
+  private createPlanApi(plan: BasePlan): Plan {
+    const createPlanRequest = this.convertToDbPlan(plan);
+    ++this.tempPlanId;
+    var planId: number = this.tempPlanId;
```

Comment:
+1 to Hana's comments. We should return an observable rather than having the subscribe here. We can also get rid of tempPlanId.